chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -191,6 +191,11 @@ pub fn pre_validate_chunk_state_witness(
 ) -> Result<PreValidationOutput, Error> {
     let store = chain.chain_store();
 
+    // Ensure that the chunk header version is supported in this protocol version
+    let protocol_version =
+        epoch_manager.get_epoch_info(&state_witness.epoch_id)?.protocol_version();
+    state_witness.chunk_header.valid_for(protocol_version)?;
```

Comment:
Sanity check - is the epoch id in the witness for the previous or the current chunk?

Reply:
I believe that it is the epoch id of the current chunk.

The comment on the witness field says:
```
/// EpochId corresponds to the next block after chunk's previous block.
    /// This is effectively the output of EpochManager::get_epoch_id_from_prev_block
    /// with chunk_header.prev_block_hash().
    /// This is needed to validate signature when the previous block is not yet
    /// available on the validator side (aka orphan state witness).
    /// TODO(stateless_validation): Deprecate once we send state witness in parts.
    pub epoch_id: EpochId,
 ```

I implemented the header version check before https://github.com/near/nearcore/pull/12307 and all the tests passed without issues. Back then only InnerV4 was allowed in the epoch that has bandwidth scheduler enabled, so I think the check would fail if it was the previous epoch.

@shreyan-gupta should be able to confirm

Reply:
Yep, it's for epoch which the state witness belongs to. We can trace the source from `ChunkStateWitness::new()`